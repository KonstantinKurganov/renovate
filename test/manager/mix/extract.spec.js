const fs = require('fs');
const path = require('path');
const { extractPackageFile } = require('../../../lib/manager/mix');

const sample = fs.readFileSync(
  path.resolve(__dirname, './_fixtures/mix.exs'),
  'utf-8'
);

describe('lib/manager/mix/extract', () => {
  describe('extractPackageFile()', () => {
    let config;
    beforeEach(() => {
      config = {};
    });
    it('returns empty for invalid dependency file', () => {
      expect(extractPackageFile('nothing here', config)).toMatchSnapshot();
    });
    it('extracts all dependencies', () => {
      const res = extractPackageFile(sample, config).deps;
      expect(res).toMatchSnapshot();
    });
  });
});
