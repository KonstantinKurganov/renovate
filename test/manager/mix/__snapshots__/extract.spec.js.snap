// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/manager/mix/extract extractPackageFile() extracts all dependencies 1`] = `
Array [
  Object {
    "currentValue": "~> 2.0.0",
    "datasource": "hex",
    "depName": "ecto",
    "depType": "hex",
    "lineNumber": 18,
  },
  Object {
    "currentValue": "~> 0.8.1",
    "datasource": "hex",
    "depName": "postgrex",
    "depType": "hex",
    "lineNumber": 19,
  },
  Object {
    "currentValue": null,
    "datasource": "github",
    "depName": "cowboy",
    "depType": "github",
    "lineNumber": 20,
    "lookupName": "ninenines/cowboy",
    "skipReason": "unsupported-version",
  },
  Object {
    "currentValue": "~> 1.0",
    "datasource": "hex",
    "depName": "secret",
    "depType": "hex",
    "lineNumber": 21,
    "organization": "acme",
  },
  Object {
    "currentValue": "~> 1.0",
    "datasource": "hex",
    "depName": "jason",
    "depType": "hex",
    "lineNumber": 22,
  },
]
`;

exports[`lib/manager/mix/extract extractPackageFile() returns empty for invalid dependency file 1`] = `
Object {
  "deps": Array [],
}
`;
